src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -136,9 +136,13 @@ private SarifLog BaselineSarifLogs(IEnumerable<Run> previous, IEnumerable<Run> c
 
         private static void AddUnmatchedResults(List<ExtractedResult> baselineResults, List<ExtractedResult> currentResults, List<MatchedResults> matchedResults)
         {
+            // Add unmatched results from Baseline log which weren't already Absent in previous run
             foreach (ExtractedResult result in baselineResults)
             {
-                matchedResults.Add(new MatchedResults(result, null));
+                if (result.Result.BaselineState != BaselineState.Absent)
```

Comment:
>Absent [](start = 65, length = 6)

Is this the right test? What if it was absent before but now it's present again?

Reply:
Then it will be regarded as new. Anything which appears that did not exist previously is new. What other state would make sense? Updated, Unchanged don't apply. You've put your finger on a core (by design) limitation in our system, we don't retain any historical information to produce a baseline state of 'Reappeared'.

This is a hard problem. On the one hand, there's the scenario of someone fixing a bug and immediately reverting it from a bad merge. In that case, it probably is the same issue (and you'd want to go find the old bug). On the other hand, results that look the same can emerge and there's no good outcome from finding an old bug that addressed the code in a different condition. I think this code is right, for now, but this conversation is likely to continue...


---
In reply to: [330294798](https://github.com/microsoft/sarif-sdk/pull/1690#discussion_r330294798) [](ancestors = 330294798)

Reply:
I think so. Basically we either filter these results before or after matching. This code does it after. If a Result was Absent in the baseline but came back, it'll match again and show up as Existing this time. Only results which remained unmatched will be filtered.

One could argue for either behavior - it might be more clear for an issue matching a previously absent one to be marked "New". However, we also see a need to match issues to ensure we update the same bug as before, which makes considering Absent results when matching a benefit.

I think we'll know more as this code gets more practical use about which way to go.